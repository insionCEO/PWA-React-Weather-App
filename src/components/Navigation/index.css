.navigation {
  align-items: center;
  display: flex;
  height: 20px;
  justify-content: center;
  margin-bottom: 50px;
  margin-top: 10px;
  width: 100%;
}

.navigation__dot {
  background-color: transparent;
  border: 1px solid #ffffff;
  border-radius: 6px;
  height: 6px;
  margin-left: 4px;
  margin-right: 4px;
  opacity: 0;
  transform: translateY(-20px);
  transition: transform .3s cubic-bezier(.165, .84, .44, 1), opacity .3s ease-out;
  width: 6px;
}

.navigation__dot.animate-in {
  opacity: 1;
  transform: translateY(0);
}
