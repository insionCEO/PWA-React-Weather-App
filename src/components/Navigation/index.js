import React, { Component } from 'react';
import rAFTimeout from '../../helpers/rAFTimeout';
import './index.scss';

class Navigation extends Component {
  constructor() {
    super();

    this.navigation = React.createRef();
  }

  animate() {
    rAFTimeout(() => {
      this.dots.map((elment, index) => {
        return rAFTimeout(() => {
          elment.classList.add('animate-in')
        }, index * 150);
      });
    }, 350);
  }

  componentDidMount() {
    this.dots = [...this.navigation.current.querySelectorAll('.navigation__dot')];

    rAFTimeout(() => this.animate(), 350);
  }

  render() {
    return (
      <section ref={this.navigation} className="navigation">
        <div className="navigation__dot"></div>
        <div className="navigation__dot"></div>
      </section>
    )
  }
}

export default Navigation;
