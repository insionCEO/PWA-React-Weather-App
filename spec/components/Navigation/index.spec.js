import React from 'react';
import Navigation from '../../../src/components/Navigation';
import { enzymeConfig, mount } from '../../enzymeConfig';

enzymeConfig();

describe('Navigation', () => {
  it('matches snapshot', () => {
    const component = mount(<Navigation />);

    expect(component).toMatchSnapshot();
  });

  describe('componentDidMount', () => {
    it('matches count of rays', () => {
      const component = mount(<Navigation />);
      const instance = component.instance();

      expect(instance.dots.length).toEqual(2);
    });

    it('calls animate', (done) => {
      const component = mount(<Navigation />);
      const instance = component.instance();

      instance.animate = jest.fn();

      setTimeout(() => {
        expect(instance.animate).toBeCalled();
        done();
      }, 400);
    });
  });
});
