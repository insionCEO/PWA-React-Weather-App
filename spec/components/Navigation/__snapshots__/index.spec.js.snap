// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation matches snapshot 1`] = `
<Navigation>
  <section
    className="navigation"
  >
    <div
      className="navigation__dot"
    />
    <div
      className="navigation__dot"
    />
  </section>
</Navigation>
`;
